datafusion/core/src/catalog/information_schema.rs

```diff
@@ -117,15 +121,19 @@ impl CatalogProvider for CatalogWithInformationSchema {
 /// providers, they will appear the next time the `information_schema`
 /// table is queried.
 struct InformationSchemaProvider {
+    config: InformationSchemaConfig,
+}
+
+#[derive(Clone)]
+struct InformationSchemaConfig {
     catalog_list: Arc<dyn CatalogList>,
     config_options: Arc<RwLock<ConfigOptions>>,
 }
 
-impl InformationSchemaProvider {
+impl InformationSchemaConfig {
     /// Construct the `information_schema.tables` virtual table
-    fn make_tables(&self) -> Arc<dyn TableProvider> {
+    fn make_tables(&self, builder: &mut InformationSchemaTablesBuilder) {
```

Comment:
The signature of these is now a little bit strange, but I wanted to try to minimise code movement

Reply:
maybe we can refactor them in a follow on PR